src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -0,0 +1,304 @@
+ * @returns {Object}
+ */
+export default function parseMetaPlaylist(
+  data : unknown,
+  url? : string
+): IParserResponse<IParsedManifest> {
+  let parsedData;
+  if (typeof data === "object" && data != null) {
+    parsedData = data;
+  } else if (typeof data === "string") {
+    try {
+      parsedData = JSON.parse(data);
+    } catch (error) {
+      throw new Error("MPL Parser: Bad MetaPlaylist file. Expected JSON.");
+    }
+  } else {
+    throw new Error("MPL Parser: Parser input must be either a string " +
+                    "or the MetaPlaylist data directly.");
+  }
+
+  const { contents, version, type } = parsedData;
+
+  if (type !== "MPL") {
+    throw new Error("MPL Parser: Bad MetaPlaylist. " +
+                    "The `type` property is not set to `MPL`");
+  }
+
+  if (typeof version !== "string" || version.split(".")[0] !== "0") {
+    throw new Error("MPL Parser: Bad MetaPlaylist version");
+  }
+
+  // quick checks
+  if (contents == null || contents.length === 0) {
+    throw new Error("MPL Parser: No content found.");
+  }
+  const ressources : Array<{ url : string; transportType : string }> = [];
+  for (let i = 0; i < contents.length; i++) {
+    const content = contents[i];
+    if (
+      content.url == null ||
+      content.startTime == null ||
+      content.endTime == null ||
+      content.transport == null
+    ) {
+      throw new Error("MPL Parser: Malformed content.");
+    }
+    ressources.push({ url: content.url, transportType: content.transport });
+  }
+
+  const metaPlaylist : IMetaPlaylist = parsedData;
+  return {
+    type : "needs-manifest-loader",
+    value : {
+      ressources,
+      continue : function parseWholeMPL(loadedRessources : Manifest[]) {
+        const parsedManifest = createManifest(metaPlaylist, loadedRessources, url);
+        return { type: "done", value: parsedManifest };
+      },
+    },
+  };
+}
+
+/**
+ * From several parsed manifests, generate a single manifest
+ * which fakes live content playback.
+ * Each content presents a start and end time, so that periods
+ * boudaries could be adapted.
+ * @param {Object} mplData
+ * @param {Array<Object>} manifest
+ * @param {string} url
+ * @returns {Object}
+ */
+function createManifest(
+  mplData : IMetaPlaylist,
+  manifests : Manifest[],
+  url? : string
+): IParsedManifest {
+  const generateAdaptationID = idGenerator();
+  const generateRepresentationID = idGenerator();
+  const { contents } = mplData;
+  const minimumTime = contents.length ? contents[0].startTime :
+                                        0;
+  const maximumTime = contents.length ? contents[contents.length - 1].startTime :
+                                        0;
+  const isLive = mplData.dynamic === true;
+  let duration : number|undefined = 0;
+
+  const periods : IParsedPeriod[] = [];
+  for (let iMan = 0; iMan < contents.length; iMan++) {
+    const content = contents[iMan];
+    const contentOffset = content.startTime;
```

Comment:
I was thinking of something.
The `startTime` is the absolute start time of a content. Then, if the content starts relatively at `T > 0`, shouldn't the contentOffset be equal to `content.startTime - T` ?
If a static base content starts at 10, and the metaplaylist declares that it should start at 1000, the content offset should be 1000 ? Or is the start time declared in the metaplaylist a _start from origin_ ?